src/response.rs

```diff
@@ -202,6 +201,51 @@ impl Response {
         self.cookie_events.push(CookieEvent::Removed(cookie));
     }
 
+    /// Returns an optional reference to an error if the response contains one.
+    pub fn error(&self) -> Option<&Error> {
+        self.res.error()
+    }
+
+    /// Returns an optional reference to a downcasted error if the response contains an error with a matching typeid.
+    ///
+    /// # Example
+    /// 
+    /// ```
+    /// # use async_std::task::block_on;
+    /// # fn main() -> Result<(), std::io::Error> { block_on(async {
+    /// #
+    /// use tide::Response;
+    ///
+    /// let error = tide::http::Error::from_str(tide::StatusCode::BadRequest, "Uh oh");
+    ///
+    /// let mut res = Response::new(400);
+    /// res.set_error(error);
+    ///
+    /// if let Some(err) = res.downcast_error::<tide::http::Error>() {
```

Comment:
I think using tide::http::Error is confusing here, because the purpose of downcasting is to yield external error types